### PR TITLE
Added Acceptance Test for Rep and Individual Journey

### PR DESCRIPTION
**Pull request by kkodalikk**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-4578

### Change description ###

Added Happy Path Acceptance Test for Rep Journey Self Test

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Constants/SelfTestPageNames.cs [hunk 1]
```diff
@@ -3,5 +3,10 @@
     public class SelfTestPageNames
     {
         public const string CheckYourComputer = "check your computer";
+        public const string UseCameraAndMicrophone = "use camera and microphone again";
```
Comment (rehmanab) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Constants/SelfTestPageNames.cs]: These have to be renamed

Comment (rehmanab) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Constants/SelfTestPageNames.cs]: Actually, you need to merge Master into this branch and you will have my changes

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature [hunk 2]
```diff
@@ -14,3 +14,34 @@ Scenario: Representative participant has camera and microphone on laptop or desk
...
+	And proceeds to next page
+	Then Representative should be on 'use camera and microphone again' screen
+	When proceeds to next page
+	Then Representative should be on 'self test' screen
+	When proceeds to next page
+	And provides answer
+	 | page                      | answer	|
+	 | camera working            | Yes		|
+	 | microphone working		 | Yes		|
+	 | see and hear video        | Yes		|
```
Comment (rehmanab) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature]: This is changed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Navigation/PageUri.cs [hunk 3]
```diff
@@ -22,5 +22,11 @@ public class PageUri
 
         //After submission of answers
         public const string CheckYourComputer = "/check-your-computer";
+        public const string UseCameraAndMicrophone = "/use-camera-microphone-again";
+        public const string SelfTest = "/self-test";
```
Comment (rehmanab) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Navigation/PageUri.cs]: these are changed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs [hunk 4]
```diff
@@ -19,7 +19,12 @@ public sealed class IndividualQuestionnaireSteps : QuestionnaireJourney
         private readonly DecisionJourney _yourInternetConnection;
         private readonly DecisionJourney _accessToRoom;
         private readonly DecisionJourney _consent;
-        private readonly JourneyStepPage _checkYourComputer;
+        private readonly JourneyStepPage _sameComputer;
+        private readonly DecisionJourney _useCameraAndMicrophone;
+        private readonly DecisionJourney _selfTest;
```
Comment (rehmanab) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs]: changed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs [hunk 5]
```diff
@@ -31,7 +36,12 @@ public IndividualQuestionnaireSteps(BrowserContext browserContext, InformationSt
             _yourInternetConnection = new DecisionJourney(browserContext, PageUri.YourInternetConnectionPage);
             _accessToRoom = new DecisionJourney(browserContext, PageUri.AccessToARoomPage);
             _consent = new DecisionJourney(browserContext, PageUri.ConsentPage);
-            _checkYourComputer = new JourneyStepPage(browserContext, PageUri.CheckYourComputer);
+            _sameComputer = new JourneyStepPage(browserContext, PageUri.SameComputer);
+            _useCameraAndMicrophone = new DecisionJourney(browserContext, PageUri.UseCameraAndMicrophone);
+            _selfTest = new DecisionJourney(browserContext, PageUri.SelfTest);
+            _cameraWorking = new DecisionJourney(browserContext, PageUri.CameraWorking);
+            _microphoneWorking = new DecisionJourney(browserContext, PageUri.MicrophoneWorking);
+            _seeAndHearVideo = new DecisionJourney(browserContext, PageUri.SeeAndHearVideo);
```
Comment (rehmanab) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs]: renamed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs [hunk 6]
```diff
@@ -128,8 +138,23 @@ public void ThenParticipantShouldProceedToPage(string page)
...
+                case "self test":
+                    _selfTest.Validate();
+                    break;
+                case "camera working":
+                    _cameraWorking.Validate();
+                    break;
+                case "microphone working":
+                    _microphoneWorking.Validate();
+                    break;
+                case "see and hear video":
```
Comment (rehmanab) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs]: here changed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/RepresentativeQuestionnaireSteps.cs [hunk 7]
```diff
@@ -212,8 +223,18 @@ private Page GetPage(string page)
...
+                    return _sameComputer;
+                case SelfTestPageNames.UseCameraAndMicrophone:
+                    return _useCameraAndMicrophone;
+                case SelfTestPageNames.SelfTest:
+                    return _selfTest;
+                case SelfTestPageNames.CameraWorking:
+                    return _cameraWorking;
+                case SelfTestPageNames.MicrophoneWorking:
+                    return _microphoneWorking;
+                case SelfTestPageNames.SeeAndHearVideo:
```
Comment (rehmanab) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/RepresentativeQuestionnaireSteps.cs]: these are renamed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs (hunk 6 above)
Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs]: Changed

Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs]: Changed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/RepresentativeQuestionnaireSteps.cs (hunk 7 above)
Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/RepresentativeQuestionnaireSteps.cs]: Changed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs (hunk 5 above)
Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs]: Changed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs (hunk 4 above)
Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs]: Changed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Navigation/PageUri.cs (hunk 3 above)
Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Navigation/PageUri.cs]: Changed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature (hunk 2 above)
Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature]: Changed

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Constants/SelfTestPageNames.cs (hunk 1 above)
Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Constants/SelfTestPageNames.cs]: Merged master into this branch and also made the required changes

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/Consent.feature [hunk 8]
```diff
@@ -10,6 +10,18 @@ Scenario: Individual participant consents to video hearing
 	When provides answer as Yes
 	When proceeds to next page
 	Then Individual should be on 'check your computer' screen
+	When provides answer as Yes
```
Comment (ddikman) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/Consent.feature]: It doesn't feel like this test maps to the scenario any more, should this entire test be on the `ThankYou` feature instead? And maybe have a different scenario text?

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature [hunk 9]
```diff
@@ -14,3 +14,34 @@ Scenario: Representative participant has camera and microphone on laptop or desk
 	Then Representative should be on 'questionnaire completed' screen
 	When proceeds to next page
 	Then Representative should be on 'check your computer' screen
+
+@smoketest @VIH-4578
+Scenario: Representative views thank you page after submitting self test questionnaire completed page for a client who is suitable for a video hearing
```
Comment (ddikman) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature]: This test has a few too many steps I think. Would it be possible to reduce them somehow and make it more to the point of what it is testing?

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature [hunk 10]
```diff
@@ -14,3 +14,34 @@ Scenario: Representative participant has camera and microphone on laptop or desk
...
+	 | access to a suitable room | Yes     |
+	 | about your client         | No     |
+	 | client attendance         | No      |
+	 | hearing suitability       | No      |
+	 | your computer             | Yes     |
+	 | about your computer       | Yes |
+
+	Then Representative should be on 'questionnaire completed' screen
+	And all the answers should match
+	And a link with text 'Print my answers' to print the answers should be visible
```
Comment (ddikman) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature]: For example, I don't think we need to test this in this test as it's covered elsewhere

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs [hunk 11]
```diff
@@ -19,7 +19,12 @@ public sealed class IndividualQuestionnaireSteps : QuestionnaireJourney
         private readonly DecisionJourney _yourInternetConnection;
         private readonly DecisionJourney _accessToRoom;
         private readonly DecisionJourney _consent;
-        private readonly JourneyStepPage _checkYourComputer;
+        private readonly DecisionJourney _checkYourComputer;
```
Comment (ddikman) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs]: As you mentioned on slack with the duplication, maybe you can move these out to a separate `SelfTestSteps` class?

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/Consent.feature (hunk 8 above)
Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/Consent.feature]: Created a ThankYou feature for Individual and moved the scenario into it

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature (hunk 9 above)
Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature]: Reduced number of tests also merged two scenarios into one

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature (hunk 10 above)
Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Features/ThankYou.feature]: Removed the step

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs (hunk 11 above)
Comment (kkodalikk) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/IndividualQuestionnaireSteps.cs]: As discussed before, we need to refactor the code in the acceptance tests to reduce duplication.

**Pull request merged**